.github/workflows/windows.yml

```diff
@@ -17,487 +17,451 @@ concurrency:
 
 env:
   PYTHON_VERSION: '3.11'
-  OV_BRANCH: 'master'
-  OV_TARBALL: ''
+  TARGET_BRANCH: 'master'
+  CMAKE_CXX_COMPILER_LAUNCHER: ccache
+  CMAKE_C_COMPILER_LAUNCHER: ccache
+  CCACHE_MAXSIZE: 500Mi
 
 jobs:
   openvino_download:
-    name: Download OpenVINO package
+    name: Download prebuilt OpenVINO
     outputs:
       status: ${{ steps.openvino_download.outcome }}
+      ov_artifact_name: ${{ steps.openvino_download.outputs.ov_artifact_name }}
+      ov_wheel_source: ${{ steps.openvino_download.outputs.ov_wheel_source }}
+      ov_version: ${{ steps.openvino_download.outputs.ov_version }}
     timeout-minutes: 10
     defaults:
       run:
         shell: bash
-    runs-on: ubuntu-22.04
+    runs-on: aks-linux-2-cores-8gb
+    container:
+      image: 'openvinogithubactions.azurecr.io/openvino_provider:0.1.0'
+      volumes:
+        - /mount:/mount
+        - ${{ github.workspace }}:${{ github.workspace }}
+    continue-on-error: true
 
     steps:
-      - name: Download OpenVINO build
-        id: openvino_download
-        run: |
-          wget ${{ env.OV_TARBALL}} --progress=bar:force:noscroll -O openvino_package.zip
-          unzip -l openvino_package.zip
-        continue-on-error: true
-
-      #
-      # Upload to artifacts
-      #
-
-      - name: Upload openvino package
-        if: steps.openvino_download.outcome == 'success'
-        uses: actions/upload-artifact@v4
-        with:
-          name: openvino_package
-          path: openvino_package.zip
-          if-no-files-found: 'error'
-
-  openvino_build:
-    name: Build OpenVINO package
-    needs: [openvino_download]
-    if: needs.openvino_download.outputs.status != 'success'
-    timeout-minutes: 150
+    - uses: openvinotoolkit/openvino/.github/actions/openvino_provider@master
+      id: openvino_download
+      with:
+        platform: 'windows'
+        commit_packages_to_provide: 'wheels'
+        revision: 'latest_available_commit'
+
+  genai_build_cpack:
+    name: genai cpack (${{ matrix.build_type }})
+    strategy:
+      matrix:
+        build_type: [Release, Debug]
+    needs: [ openvino_download ]
+    timeout-minutes: 60
     defaults:
       run:
         shell: pwsh
-    runs-on: windows-2022-16-core
+    runs-on: windows-2022
```

Comment:
should not we use self hosted runner?

Reply:
should, I'll switch on them when all the jobs are passed